### ISSUE TITLE
File in use error (backup and upload work)

### ISSUE THREAD

**Issue by gitproos** [lavinir/hassio-onedrive-backup#78]
The backups are working, but I'm reading errors in the logs. Specifically: "The process cannot access the file '/tmp/a13b55bb.tar' because it is being used by another process.". Maybe the script tries to upload too soon while compression/packing is still in progress? The backups are a bit large, about 2,5GB. I have reduced in size already from 4GB files, but the errors remain.

```
04/11/2023 02:47:17 [24] Info: Creating new backup
04/11/2023 02:47:17 [24] Info: Starting partial local backup
04/11/2023 02:52:17 [12] Info: Creating new backup
04/11/2023 02:52:17 [12] Info: Starting partial local backup
04/11/2023 02:52:17 [12] Info: Backup complete
04/11/2023 02:57:18 [21] Info: Creating new backup
04/11/2023 02:57:18 [21] Info: Starting partial local backup
04/11/2023 02:57:18 [20] Info: Backup complete
04/11/2023 03:02:18 [21] Info: Creating new backup
04/11/2023 03:02:18 [21] Info: Starting partial local backup
04/11/2023 03:02:18 [21] Info: Backup complete
04/11/2023 03:07:20 [17] Info: Creating new backup
04/11/2023 03:07:20 [17] Info: Starting partial local backup
04/11/2023 03:07:20 [17] Info: Backup complete
04/11/2023 03:10:57 [12] Info: Backup complete
04/11/2023 03:12:20 [20] Info: Found 1 backups to upload.
04/11/2023 03:12:20 [21] Info: Uploading hass_backup_2023-04-11-02-47 (04/11/2023 02:47:17)
04/11/2023 03:12:20 [21] Info: Fetching Local Backup (Slug:a13b55bb)
04/11/2023 03:12:23 [21] Info: Backup (a13b55bb) fetched successfully
04/11/2023 03:12:23 [21] Info: Starting file upload. (Size:2161111040 bytes. Attempt: 1/3)
04/11/2023 03:17:21 [15] Info: Found 1 backups to upload.
04/11/2023 03:17:21 [15] Info: Uploading hass_backup_2023-04-11-02-47 (04/11/2023 02:47:17)
04/11/2023 03:17:21 [15] Info: Fetching Local Backup (Slug:a13b55bb)
04/11/2023 03:17:21 [15] Error: Error uploading backup: System.IO.IOException: The process cannot access the file '/tmp/a13b55bb.tar' because it is being used by another process.
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Init(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Int64& fileLength, UnixFileMode& filePermissions)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String fullPath, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, UnixFileMode openPermissions, Int64& fileLength, UnixFileMode& filePermissions, Func`4 createOpenException)
   at System.IO.Strategies.OSFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options, Int64 preallocationSize, Nullable`1 unixCreateMode)
   at System.IO.File.Create(String path)
   at hassio_onedrive_backup.Hass.HassioClient.DownloadBackupAsync(String backupSlug) in /src/Hass/HassioClient.cs:line 197
   at hassio_onedrive_backup.Hass.HassioClient.DownloadBackupAsync(String backupSlug) in /src/Hass/HassioClient.cs:line 200
   at hassio_onedrive_backup.Hass.BackupManager.UploadLocalBackupToOneDrive(Backup backup, Action`2 progressCallback, Boolean updateHassEntityState) in /src/Hass/BackupManager.cs:line 228
04/11/2023 03:17:22 [15] Info: Removing 1 local backups
04/11/2023 03:19:18 [21] Info: Upload completed successfully
04/11/2023 03:19:19 [21] Info: Found 1 backups to delete from OneDrive.
04/11/2023 03:19:19 [21] Info: Deleting item: hass_backup_2023-03-27-02-40.tar
04/11/2023 03:19:19 [21] Info: Removing 1 local backups
```

**Comment by lavinir** [comment]
Hi, thank you for sending.
This is a bug - looks like next backup iteration is starting while the previous one is still executing. I will publish a minor fix for this.

**Comment by lavinir** [comment]
Should be fixed now in 2.1.1

**Comment by gitproos** [comment]
Updated, tested and can report no more errors in the logs. Thanks!
